Slim/Control/Request.pm

```diff
@@ -2398,16 +2398,19 @@ sub __matchingRequest {
 # return compiled regexp representing the $possibleNames array of arrays
 sub __requestRE {
 	my $possibleNames = shift || return qr /./;
-	my $regexp = '';
-
-	my $i = 0;
+	my @regexp = ();
 
 	for my $names (@$possibleNames) {
-		$regexp .= ',' if $i++;
-		$regexp .= (scalar @$names > 1) ? '(?:' . join('|', @$names) . ')' : $names->[0];
+
+		# Bracket each name using word boundaries to avoid
+		# "play" matching "play", "playlist", and "display".
+
+		push @regexp,
+			'(?:' . join('|', map { "\\b$_\\b" } @$names) . ')'
+			if scalar @$names;
```

Comment:
If an `if` becomes multi-line I prefer to have it wrapping the conditional statement. Could you please change this?

Is the double backslash in `"\\b"` required?

Reply:
TL;DR `"\b"` is interpreted as a backspace (BS) whereas `"\\b" eq '\b' and "\\b" eq '\\b'`.

```
$pattern1 = '\\bbear';
$pattern2 = '\bbear';
$pattern3 = "\\bbear";
$pattern4 = "\bbear";

die unless
    $pattern1 eq $pattern2 &&
    $pattern1 eq $pattern3 &&
    $pattern1 ne $pattern4;;

print ' ' . ord($_) . ' ' . $_ for split(//, $pattern1);
print "\n";

print ' ' . ord($_) . ' ' . $_ for split(//, $pattern4);
print "\n";

$re1 = qr /$pattern1/;
$re4 = qr /$pattern4/;

$candidate = "black-bear";

die unless $candidate =~ $re1;
die if $candidate =~ $re4;
```